### PR TITLE
ENG-4402 replace with formik

### PR DESCRIPTION
**Pull request by davdet**
Preferences form replaced with Formik

### CODE REVIEW

File: src/ui/users/my-profile/AppSettingsForm.js [hunk 1]
```diff
@@ -28,120 +33,166 @@ export class AppSettingsFormBody extends Component {
...
-      PropTypes.number,
-    ]),
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     text: PropTypes.string,
   })).isRequired,
-  defaultPageJoinGroups: PropTypes.arrayOf(PropTypes.string).isRequired,
-  defaultContentJoinGroups: PropTypes.arrayOf(PropTypes.string).isRequired,
+  // defaultPageJoinGroups: PropTypes.arrayOf(PropTypes.string).isRequired,
+  // defaultContentJoinGroups: PropTypes.arrayOf(PropTypes.string).isRequired,
 };
```
Comment (ichalagashvili) [inline src/ui/users/my-profile/AppSettingsForm.js]: please remove commented lines

File: src/ui/users/my-profile/AppSettingsForm.js [hunk 2]
```diff
@@ -28,120 +33,166 @@ export class AppSettingsFormBody extends Component {
...
+        </Form>
+      </FormikForm>
     );
   }
 }
 
 AppSettingsFormBody.propTypes = {
   onDidMount: PropTypes.func.isRequired,
-  handleSubmit: PropTypes.func.isRequired,
+  // handleSubmit: PropTypes.func.isRequired,
```
Comment (ichalagashvili) [inline src/ui/users/my-profile/AppSettingsForm.js]: please remove comment

File: src/ui/users/my-profile/AppSettingsForm.js [hunk 3]
```diff
@@ -28,120 +33,166 @@ export class AppSettingsFormBody extends Component {
 
   render() {
     const {
-      selectGroups, defaultPageJoinGroups, defaultContentJoinGroups,
+      selectGroups,
+      // defaultPageJoinGroups,
+      // defaultContentJoinGroups,
+      // values,
```
Comment (ichalagashvili) [inline src/ui/users/my-profile/AppSettingsForm.js]: please remove comments

**Review by ichalagashvili — CHANGES_REQUESTED** [review]
comments need to be removed

File: src/ui/users/my-profile/AppSettingsForm.js (hunk 1 above)
Comment (davdet) [reply src/ui/users/my-profile/AppSettingsForm.js]: done!

File: src/ui/users/my-profile/AppSettingsForm.js (hunk 2 above)
Comment (davdet) [reply src/ui/users/my-profile/AppSettingsForm.js]: done!

File: src/ui/users/my-profile/AppSettingsForm.js (hunk 3 above)
Comment (davdet) [reply src/ui/users/my-profile/AppSettingsForm.js]: done!

**Review by ichalagashvili — APPROVED** [review]

**Pull request merged**